doc/nrf/config_and_build/bootloaders_and_dfu/bootloader.rst

```diff
@@ -106,18 +108,46 @@ If the verification fails, the boot process stops.
 
-Except for providing your own keys, there is no need to modify the immutable bootloader in any way before you program it.
-The verification provided by this bootloader is recommended and suitable for all the most common user scenarios and includes the following checks:
+1. Locking of the flash memory.
 
-* Signature verification - Verifies that the key used for signing the next image in the boot sequence matches one of the provided public keys.
-* Metadata verification - Checks that the images are compatible.
+   To enable the RoT, the immutable bootloader locks the flash memory address range containing itself and its configuration using the hardware available on the given architecture.
+   (The immutable bootloader cannot be modified or deleted without erasing the entire device.)
 
-.. caution::
-   You must generate and use your own signing keys while in development and before deploying when using either MCUboot or the |NSIB| as an immutable bootloader.
-   See :ref:`ug_fw_update_development_keys` for more information.
+#. Selection of the next slot in the boot chain.
+
+   The next stage in the boot chain can either be an application or another bootloader:
+
+   * When the next stage is an application, you can allocate one or two slots for it.
+   * When the next stage is a second-stage bootloader, two slots are always used.
+
+   Each slot has a version number, and the bootloader will select the slot with the latest version.
+   For more information about creating a second stage bootloader, see :ref:`ug_bootloader_adding_upgradable`.
+
+#. Verification of the next stage in the boot chain.
+
+   The verification provided by this bootloader is recommended and suitable for all the most common user scenarios and includes the following checks:
+
+   * Signature verification - Verifies that the key used for signing the next image in the boot sequence matches one of the provided public keys.
+
+     The next image has metadata containing the full public key that corresponds to the private key used to sign the firmware.
+     This public key is checked against the provisioned hashes of public keys to determine if the image is valid.
+
+     All public key hashes at lower indices than the matching hash are permanently invalidated at this point.
+     You can use this mechanism to decommission broken keys.
+
+   * Metadata verification - Checks that the images are compatible.
+
+   .. caution::
+      You must generate and use your own signing keys while in development and before deploying when using either MCUboot or the |NSIB| as an immutable bootloader.
+      See :ref:`ug_fw_update_development_keys` for more information.
+
+#. Booting of the next stage in the boot chain.
+
+   All peripherals that have been used are reset and the next stage is booted.
+
+Except for providing your own keys, there is no need to modify the immutable bootloader in any way before you program it.
```

Comment:
> there is no need to modify the immutable bootloader in any way before you program it.
Something seems off with this statement.
I agree that you do not have to do this. 
However, in several situations, it is beneficial to configure the immutable bootloader. To mention one: If you want MCUboot as an immutable bootloader with Serial Recovery enabled, you must configure it.